src/main/webapp/app/modules/provider/record/record-create.tsx

```diff
@@ -545,139 +585,146 @@ export class RecordCreate extends React.Component<IRecordCreateViewProp, IRecord
+                  const isPhoneValid =
+                    _.get(this.state.services, `[${i}].phones[0].number`, '') &&
+                    !isPossiblePhoneNumber(_.get(this.state.services, `[${i}].phones[0].number`, ''));
+                  return (
+                    <Row key={`service-${i}`} className="item service">
+                      <Col md={1}>
+                        <h4>{i + 1}.</h4>
+                      </Col>
+                      <Col md={11}>
+                        <AvGroup className="flex">
+                          <div className="required" />
+                          <Label className="sr-only" for={'service-id[' + i + '].name'}>
+                            {translate('entity.validation.required')}
+                          </Label>
+                          <AvInput
+                            id={'service-id[' + i + '].name'}
+                            type="text"
+                            name={'services[' + i + '].name'}
+                            placeholder={translate('record.service.name')}
+                            validate={{
+                              required: { value: true, errorMessage: translate('entity.validation.required') }
+                            }}
+                            onChange={this.onServiceChange(i, 'name')}
+                          />
+                        </AvGroup>
+                        <AvGroup className="flex">
+                          <div className="required" />
+                          <Label className="sr-only" for={'services[' + i + '].taxonomyIds'}>
+                            {translate('record.service.type')}
+                          </Label>
+                          <AvSelect
+                            name={'services[' + i + '].taxonomyIds'}
+                            validate={{
+                              required: { value: true, errorMessage: translate('entity.validation.required') }
+                            }}
+                            options={taxonomyOptions}
+                            // @ts-ignore
+                            isMulti
+                            placeholder={translate('record.service.type')}
+                            onChange={this.onServiceChange(i, 'taxonomyIds')}
+                          />
+                        </AvGroup>
+                        <div className={`record-create ${isPhoneValid ? 'invalid' : ''}`}>
+                          <Input
+                            className="my-2 form-control"
+                            type="text"
+                            name="phone"
+                            id={'service-id[' + i + '].phone'}
+                            placeholder={isMobile ? translate('referral.placeholder.phoneMobile') : translate('referral.placeholder.phone')}
+                            onChange={this.setServicePhone(i)}
+                            value={_.get(this.state.services, `[${i}].phones[0].number`, '')}
```

Comment:
I wonder if using the Phone entity is the way to go here - if I remember correctly, it has separate 'extension' and 'number' fields, and we are saving both extension and phone number in the 'number' field here

Reply:
on top of that, we are using only the first phone number from the list, is the order guaranteed? what would happen if a record has multiple phone numbers

Reply:
perhaps we could have 'phoneNumber' field in dtos and handle saving and retrieving it on the backend, or have a separate 'phoneNumber' field in the entity and migrate the ones from old view when needed

Reply:
I think we should follow the existing architecture whenever we can, but I'll leave the decision to @kdondziak which solution he prefers in this case

Reply:
@jkondrat actually, you are right using the existing phone entity brings a lot of issues.